app/models/item.rb

```diff
@@ -1,10 +1,34 @@
 class Item < ActiveRecord::Base
-  validates :name, :description, :price, :seller_id, presence: true
+  validates :name, :description, :seller_id, presence: true
+  validate :price, :auction_details, :auction_duration
 
   scope :available, -> { where(available: true) }
 
   belongs_to :seller, class_name: "User", foreign_key: :seller_id
   has_one :purchaser, through: :purchases, source: :purchaser
+  has_many :bids
 
   mount_uploader :image, ImageUploader
+
+  def highest_bid
+    bids.maximum(:amount) || starting_bid_price
+  end
+
+  def price
+    if !buy_it_now_price && !starting_bid_price
+      errors.add(:price, "you need to select either a starting bid or a buy it now price")
```

Comment:
I'd prefer to see these message cleaned up to use proper capitalization and punctuation. Ideally you'd use [internationalization](http://guides.rubyonrails.org/i18n.html) but I'm ok without it for now.